### PR TITLE
artificial delay before redirect to checkin #86

### PR DESCRIPTION
**Pull request by sakurano92**
simple timer added need to work on but will delay for 10 seconds for now
still need to add following:
1. separate that delay to single class so can be implemented whenever needed delay (i need help for this)
2. have to show on UI that it will redirect with in 10 seconds... (new html? or just show on same page?)

### CODE REVIEW

**Review by dubeme — CHANGES_REQUESTED** [review]
Nice start, but will need some adjustment.

Remember that the tasks asks for delaying before redirecting. Currently you're delaying before taking any action at all.

You should instead delay before the call to "RedirectToAction".

File: app/wama.web/Controllers/PatronController.cs [hunk 1]
```diff
@@ -23,8 +23,9 @@ public PatronController(IUserAccountService userAccountService, ICheckInService
         }
 
         [HttpGet]
-        public IActionResult Create()
+        public IActionResult Create(string ReceavedId)
```
Comment (dubeme) [inline app/wama.web/Controllers/PatronController.cs]: I'm guessing "**ReceavedId**" is supposed to be "**receivedId**".

**Apart** from that, you should rename the variable to "**memberId**", this tells me when reading the code that the parameter is a Member Id.

File: app/wama.web/Controllers/PatronController.cs [hunk 2]
```diff
@@ -23,8 +23,9 @@ public PatronController(IUserAccountService userAccountService, ICheckInService
...
         {
+            ViewBag.MemberId = ReceavedId;
```
Comment (dubeme) [inline app/wama.web/Controllers/PatronController.cs]: Also here

File: app/wama.web/Controllers/CheckInController.cs [hunk 3]
```diff
@@ -33,7 +33,7 @@ public IActionResult Index(string memberId)
                     return RedirectToAction(
                         actionName: nameof(PatronController.Create),
                         controllerName: nameof(PatronController).Replace(AppString.Controller, string.Empty),
-                        routeValues: memberId);
+                        routeValues: new { ReceavedId = memberId });
```
Comment (dubeme) [inline app/wama.web/Controllers/CheckInController.cs]: See comment below about renaming the variable

**Review by dubeme — CHANGES_REQUESTED** [review]
Almost there, see comments on minor changes to make.

File: app/wama.web/Controllers/CheckInController.cs (hunk 3 above)
Comment (sakurano92) [reply app/wama.web/Controllers/CheckInController.cs]: so fix the naming from ReceavedId to memberId will be all?

**Review by dubeme — CHANGES_REQUESTED** [review]
Nice work, Sang Won

**Review by dubeme — APPROVED** [review]

**Pull request merged**